lighthouse-core/lib/lh-error.js

```diff
@@ -70,128 +103,136 @@ const ERRORS = {
     lhrRuntimeError: true,
   },
   INVALID_SPEEDLINE: {
     code: 'INVALID_SPEEDLINE',
-    message: strings.didntCollectScreenshots,
+    message: UIStrings.didntCollectScreenshots,
     lhrRuntimeError: true,
   },
 
   // Trace parsing errors
   NO_TRACING_STARTED: {
     code: 'NO_TRACING_STARTED',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_NAVSTART: {
     code: 'NO_NAVSTART',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_FCP: {
     code: 'NO_FCP',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_DCL: {
     code: 'NO_DCL',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
     lhrRuntimeError: true,
   },
   NO_FMP: {
     code: 'NO_FMP',
-    message: strings.badTraceRecording,
+    message: UIStrings.badTraceRecording,
   },
 
   // TTI calculation failures
-  FMP_TOO_LATE_FOR_FCPUI: {code: 'FMP_TOO_LATE_FOR_FCPUI', message: strings.pageLoadTookTooLong},
-  NO_FCPUI_IDLE_PERIOD: {code: 'NO_FCPUI_IDLE_PERIOD', message: strings.pageLoadTookTooLong},
-  NO_TTI_CPU_IDLE_PERIOD: {code: 'NO_TTI_CPU_IDLE_PERIOD', message: strings.pageLoadTookTooLong},
+  FMP_TOO_LATE_FOR_FCPUI: {code: 'FMP_TOO_LATE_FOR_FCPUI', message: UIStrings.pageLoadTookTooLong},
+  NO_FCPUI_IDLE_PERIOD: {code: 'NO_FCPUI_IDLE_PERIOD', message: UIStrings.pageLoadTookTooLong},
+  NO_TTI_CPU_IDLE_PERIOD: {code: 'NO_TTI_CPU_IDLE_PERIOD', message: UIStrings.pageLoadTookTooLong},
   NO_TTI_NETWORK_IDLE_PERIOD: {
     code: 'NO_TTI_NETWORK_IDLE_PERIOD',
-    message: strings.pageLoadTookTooLong,
+    message: UIStrings.pageLoadTookTooLong,
   },
 
   // Page load failures
   NO_DOCUMENT_REQUEST: {
     code: 'NO_DOCUMENT_REQUEST',
-    message: strings.pageLoadFailed,
+    message: UIStrings.pageLoadFailed,
     lhrRuntimeError: true,
   },
-  /* Used when DevTools reports loading failed. Usually an internal (Chrome) issue. */
+  /* Used when DevTools reports loading failed. Usually an internal (Chrome) issue.
+   * Requries an additional `errorDetails` field for translation.
+   */
   FAILED_DOCUMENT_REQUEST: {
     code: 'FAILED_DOCUMENT_REQUEST',
-    message: strings.pageLoadFailed,
+    message: UIStrings.pageLoadFailedWithDetails,
     lhrRuntimeError: true,
   },
-  /* Used when status code is 4xx or 5xx. */
+  /* Used when status code is 4xx or 5xx.
+   * Requires an additional `statusCode` field for translation.
+   */
   ERRORED_DOCUMENT_REQUEST: {
     code: 'ERRORED_DOCUMENT_REQUEST',
-    message: strings.pageLoadFailed,
+    message: UIStrings.pageLoadFailedWithStatusCode,
     lhrRuntimeError: true,
   },
-  /* Used when security error prevents page load. */
+  /* Used when security error prevents page load.
+   * Requires an additional `securityMessages` field for translation.
```

Comment:
I feel like we're going to forget to update these if they ever change, but I'm not sure what to do about it

Reply:
Well the current plan is that if we change them, then there will be a ton of errors thrown from str_ because it is missing the required substitution fields.  But I'm not sure that's a long term strategy.